File: shoe-app/src/main/java/com/shoeapp/service/CartService.java
package com.shoeapp.service;

import com.shoeapp.dto.request.CartItemRequest;
import com.shoeapp.dto.response.CartItemResponse;
import com.shoeapp.dto.response.CartResponse;

public interface CartService {

    public CartResponse getCart();
    public CartItemResponse addToCart(CartItemRequest cartItemRequest);
    public CartItemResponse updateCartItem(Long itemId, CartItemRequest cartItemRequest);
    public void removeFromCart(Long itemId);
    public void clearCart();


}

File: shoe-app-frontend/src/services/cart.service.js
// src/services/cart.service.js
import api from './api';

const CartService = {
  getCart: async () => {
    const response = await api.get('/cart');
    return response.data;
  },
  
  addToCart: async (shoeId, quantity) => {
    const response = await api.post('/cart', { shoeId, quantity });
    return response.data;
  },
  
  updateCartItem: async (itemId, quantity) => {
    const response = await api.put(`/cart/${itemId}`, { quantity });
    return response.data;
  },
  
  removeFromCart: async (itemId) => {
    return api.delete(`/cart/${itemId}`);
  },
  
  clearCart: async () => {
    return api.delete('/cart');
  }
};

export default CartService;